addons/mail/static/src/models/thread/thread.js

```diff
@@ -1363,6 +1363,46 @@ function factory(dependencies) {
             return [['replace', this.activities.filter(activity => activity.state === 'overdue')]];
         }
 
+        /**
+         * Current partner seen infos.
+         *
+         * @private
+         * @returns {[mail.partner_seen_infos]}
+         */
+        _computeCurrentPartnerSeenInfos() {
+            if (!this.messaging) {
+                return [['unlink-all']];
+            }
+            if (!this.messaging.currentPartner) {
+                return [['unlink-all']];
+            }
+            return [['replace', this.partnerSeenInfos.filter(partnerSeenInfo => partnerSeenInfo.partner === this.messaging.currentPartner)]]
+        }
+
+        /**
+         * Not a real field, used to trigger `markAsFetch` when new message(s) is(are) fetched
+         *
+         * @private
+         */
+        _computeShouldBeSetAsFetched() {
+            if (this.model !== 'mail.channel') {
+                return;
+            }
+            if (this.channel_type === 'channel') {
+                return;
+            }
+            if (!this.mainCacheIsLoaded) {
+                return;
+            }
+            const lastFetchedMessage = this.currentPartnerSeenInfos.length > 0 &&
+                this.currentPartnerSeenInfos[0] &&
+                this.currentPartnerSeenInfos[0].lastFetchedMessage;
+            if (this.lastMessage && (!lastFetchedMessage || this.lastMessage.id > lastFetchedMessage.id))
```

Comment:
keep using preventive returns instead of having a complex condition here
(have last line of the method be `this.markAsFetched();` without any indent level)